File: Fib.cpp
// Filename: Fib.cpp
#include <iostream>
using namespace std;

int fibCallsCounter; // Global variable

int fib(int n) {     // Global function
  fibCallsCounter++;
  return (n <= 2) ? 1 : fib(n-1) + fib(n-2);  
}

int main(int argc, char* argv[]) {
  while(true) {
    int n;
    fibCallsCounter = 0; // Refer to global variable
    cout << "Desired value of n (0 to quite): ";
    cin >> n;    
    if(n == 0)
      break;
    // fib(n) refers to a global function call
    cout << "The " << n << "th fibonacci number is " << fib(n) << endl;
    cout << "Function was called " << fibCallsCounter << " times" << endl;
  }
}

File: src/Fib.java
// Filename: Fib.java with similar behavior to Fib.cpp
import java.util.Scanner;
import java.io.IOException;

public class Fib {
  private static int fibCallsCounter;
  private static int fib(int n) {
    fibCallsCounter++;
    return (n <= 2) ? 1 : fib(n-1) + fib(n-2);
  }
  public static void main(String[] args) throws IOException {
    Scanner sc = new Scanner(System.in);
    while(true) {
      int n;
      fibCallsCounter = 0;
      System.out.print("Desired value of n (0 to quite): ");
       n = sc.nextInt();
      if(n == 0)
        break;
      System.out.println("The " + n + "th fibonacci number is " + fib(n));
      System.out.println("Function was called " + fibCallsCounter + " times");
    }
    sc.close();
  }
}